projects/components/src/stepper/stepper.component.ts

```diff
@@ -104,26 +105,54 @@ export class StepperComponent implements AfterContentInit {
 
   public steps$?: Observable<StepperTabComponent[]>;
 
-  public ngAfterContentInit(): void {
-    this.steps$ = queryListAndChanges$(this.steps).pipe(map(list => list.toArray()));
-  }
+  private readonly isNextDisabledSubject: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(false);
+  public isNextDisabled$: Observable<boolean> = this.isNextDisabledSubject.asObservable();
 
-  // TODO: can be made a directive
-  public isNextDisabled(stepper: MatStepper): boolean {
-    const selectedStep = stepper.steps.get(stepper.selectedIndex);
-    const formValid = selectedStep?.stepControl?.valid;
-    const stepCompleted = selectedStep?.completed;
-
-    return this.isLinear && !stepCompleted && !formValid;
-  }
+  public constructor(private readonly subscriptionLifecycle: SubscriptionLifecycle) {}
 
-  // TODO: can be made a directive
-  public isSubmitDisabled(stepper: MatStepper): boolean {
-    const selectedStep = stepper.steps.get(stepper.selectedIndex);
-    const formValid = selectedStep?.stepControl?.valid;
-    const stepCompleted = selectedStep?.completed;
+  public ngAfterContentInit(): void {
+    this.steps$ = queryListAndChanges$(this.steps).pipe(map(list => list.toArray()));
 
-    return !stepCompleted && !formValid;
+    /**
+     * Inorder to get the next/submit button disabled status, we need to listen to the primary triggers for change:
+     * 1. Stepper Initialization
+     * 2. Selection Change
+     *
+     * When either of these happens, we need to check if a step control is provided for the step. If a step control is provided, we
+     * switch to the form status changes inorder to find if the form is valid or not.
+     *
+     * If there is no step control provided, we can check for the step completed property.
+     *
+     * Since the same observable is used for both the disabled handling of submit and the next button,
+     * we switch between two logic:
+     * - For next: If the stepper is linear and the step is valid
+     * - For Submit: If all the steps are valid
+     */
+    const sub = merge(
+      this.selectionChange.pipe(map(({ selectedStep }) => selectedStep)),
+      this.steps$.pipe(map(([firstStep]) => firstStep))
+    )
+      .pipe(
+        switchMap(step => {
+          if (step.stepControl) {
+            return step.stepControl.statusChanges.pipe(
+              startWith(step.stepControl.status),
+              map(status => status === 'VALID')
```

Comment:
define a meaningful const value for this string.

Reply:
also, is there a value we can look for instead of doing a string comparison?

Reply:
```
type FormControlStatus = 'VALID' | 'INVALID' | 'PENDING' | 'DISABLED';
```

This is what angular uses.

Reply:
should not be more useful to have an enum in this case?

Reply:
Yeah totally. But I didn't want to define a separate enum for a framework-defined type as we have to keep parity with it in the long run.